lib/Twig/Environment.php

```diff
@@ -117,6 +119,40 @@ public function __construct(Twig_LoaderInterface $loader = null, $options = arra
     }
 
     /**
+     * Returns deprecations for templates contained in a directory.
+     *
+     * @param string $dir The directory where templates are stored
+     * @param string $ext Limit the loaded templates by extension
+     *
+     * @return An array of deprecations
+     */
+    public function getDeprecations($dir, $ext = '.twig')
+    {
+        $deprecations = array();
+        $previous = set_error_handler(function ($type, $msg) use (&$deprecations) {
+            if (E_USER_DEPRECATED === $type) {
+                $deprecations[] = $msg;
+            }
+        });
+
+        foreach (new RecursiveIteratorIterator(new RecursiveDirectoryIterator($dir), RecursiveIteratorIterator::LEAVES_ONLY) as $file) {
+            if (!preg_match('{'.preg_quote($ext).'$}', $file)) {
+                continue;
+            }
+
+            try {
+                $this->compileSource(file_get_contents($file), (string) $file);
+            } catch (Twig_Error_Syntax $e) {
+                // ignore templates containing syntax errors
+            }
+        }
+
+        set_error_handler($previous);
```

Comment:
this should use ``restore_error_handler`` instead to restore it, instead of readding the previous one on top of the stack